src/app/shared/entry.ts

```diff
@@ -202,10 +203,26 @@ export abstract class Entry implements OnInit, OnDestroy {
       if (foundVersion) {
         return foundVersion;
       }
+    } else {
+      return null;
     }
 
     // Select newest last_modified version, if it's the same, choose the top
-    return versions.reduce((a, b) => b.last_modified > a.last_modified ? b : a);
+    //  return versions.reduce((a, b) => b.last_modified > a.last_modified ? b : a);
+  }
+
+  selectTag(versions: Array<Tag>, urlVersion: string, defaultVersion: string): Tag {
+    const selectedTag = this.selectVersion(versions, urlVersion, defaultVersion);
+    if (selectedTag != null) {
```

Comment:
This is more of a preference, but I would do this using JavaScript truthiness:

`return selectedTag || versions.reduce((a,b) =>...`